common/aws/s3/s3.go

```diff
@@ -2,10 +2,47 @@ package s3
 
 import "context"
 
+// Client encapsulates the functionality of an S3 client.
 type Client interface {
-	CreateBucket(ctx context.Context, bucket string) error
+
+	// DownloadObject downloads an object from S3.
 	DownloadObject(ctx context.Context, bucket string, key string) ([]byte, error)
+
+	// UploadObject uploads an object to S3.
 	UploadObject(ctx context.Context, bucket string, key string, data []byte) error
+
+	// DeleteObject deletes an object from S3.
 	DeleteObject(ctx context.Context, bucket string, key string) error
+
+	// ListObjects lists all objects in a bucket with the given prefix. Note that this method may return
+	// file fragments if the bucket contains files uploaded via FragmentedUploadObject.
 	ListObjects(ctx context.Context, bucket string, prefix string) ([]Object, error)
+
+	// CreateBucket creates a bucket in S3.
+	CreateBucket(ctx context.Context, bucket string) error
+
+	// FragmentedUploadObject uploads a file to S3. The fragmentSize parameter specifies the maximum size of each
+	// file uploaded to S3. If the file is larger than fragmentSize then it will be broken into
+	// smaller parts and uploaded in parallel. The file will be reassembled on download.
+	//
+	// Note: if a file is uploaded with this method, only the FragmentedDownloadObject method should be used to
+	// download the file. It is not advised to use DeleteObject on files uploaded with this method (if such
+	// functionality is required, a new method to do so should be added to this interface).
```

Comment:
Interesting, how do we garbage collect this stuff after 2 weeks then? Or is the idea that we don't?

Reply:
The current use case for this method will use a bucket that enforces a TTL for all files inside of it, so we won't have to do any sort of manual deletion. Straight forward to add methods for deleting fragmented files, but my instinct is to avoid adding code that I don't have an obvious current use for.

Reply:
Agree with not adding a method that we don't currently need! Maybe document the TTL behavior here though?

Reply:
Added the following:

```
	// Note: if this operation fails partway through, some file fragments may have made it to S3 and others may not.
	// In order to prevent long term accumulation of fragments, it is suggested to use this method in conjunction with
	// a bucket configured to have a TTL.
```